### PR TITLE
[selenium 4]appium mocha samples

### PR DESCRIPTION
**Pull request by innovater21**
# PR1
- This PR consist of sample appium test using nodejs mocha in selenium-4
- https://browserstack.atlassian.net/browse/AAP-10423

### CODE REVIEW

File: android/conf/local.conf.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    'bstack:options' : {
+        "projectName" : "Mocha Android Test Project",
+        "buildName" : "browserstack-build-local",
+        "sessionName" : "local_test",
+        "userName" : USERNAME,
+        "accessKey" : ACCESS_KEY,
+        "debug" : true,
+        "local" : true
+    }
+};
```
Comment (innovater21) [inline android/conf/local.conf.js]: new line

File: android/conf/single.conf.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+USERNAME =  process.env.BROWSERSTACK_USERNAME || 'BROWSERSTACK_USERNAME',
+ACCESS_KEY = process.env.BROWSERSTACK_ACCESS_KEY || 'BROWSERSTACK_ACCESS_KEY',
+  
+exports.capabilities = {
+    "platformName" : "android",
+    "appium:deviceName" : "Google Pixel 6 Pro",
+    "browserName" : "chrome",
```
Comment (innovater21) [inline android/conf/single.conf.js:7]: do we need browserName caps ?

File: android/conf/parallel.conf.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "platformName" : "android",
+        "appium:app" : "bs://<app-id>",
+        'bstack:options' : {
+            "sessionName" : "Bstack parallel mocha",
+        }
+      },
+      {
+        "appium:deviceName" : "Google Pixel 6 Pro",
+        "appium:osVersion": "12.0",
+        "browserName" : "chrome",
```
Comment (innovater21) [inline android/conf/parallel.conf.js:33]: same

File: android/conf/single.conf.js (hunk 2 above)
Comment (avinash-bharti) [reply android/conf/single.conf.js:7]: sadly yes, to use selenium webdriver.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,130 @@
...
+### Run your first test :
+
+**1. Upoad your Android or iOS App**
+
+Upload your Android app (.apk or .aab file) or iOS app (.ipa file) to BrowserStack servers using our REST API. Here is an example cURL request :
+
+```
+curl -u "YOUR_USERNAME:YOUR_ACCESS_KEY" \
+-X POST "https://api-cloud.browserstack.com/app-automate/upload" \
+-F "file=@/path/to/apk/file"
```
Comment (rushi04) [inline README.md:43]: is upload via url needed ?

File: android/conf/local.conf.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+USERNAME =  process.env.BROWSERSTACK_USERNAME || 'BROWSERSTACK_USERNAME',
+ACCESS_KEY = process.env.BROWSERSTACK_ACCESS_KEY || 'BROWSERSTACK_ACCESS_KEY',
+  
+exports.capabilities = {
+    "platformName" : "android",
+    "appium:deviceName" : "Samsung Galaxy S22 Ultra",
+    "browserName" : "chrome",
```
Comment (rushi04) [inline android/conf/local.conf.js:7]: ??

File: android/conf/single.conf.js (hunk 2 above)
Comment (rushi04) [reply android/conf/single.conf.js:7]: Possible to explain ? Do we pass both app id and browser combined for bot appium  ?

File: README.md (hunk 4 above)
Comment (avinash-bharti) [reply README.md:43]: You mean, dashboard upload should be here?
Rest of the sample repos suggests to use rest api to upload the app.

File: android/conf/local.conf.js (hunk 5 above)
Comment (avinash-bharti) [reply android/conf/local.conf.js:7]: - For nodejs/mocha we had two options of solutions that is built for appium i.e webdriverio and wd.
- We already have webdriverio sample repo and wd is deprecated.
- Alternative is to use selenium driver only and for that we have to use the browser cap (limitation). This same set of caps are used in node sdk too.

File: android/conf/parallel.conf.js (hunk 3 above)
Comment (avinash-bharti) [reply android/conf/parallel.conf.js:33]: https://github.com/browserstack/mocha-appium-app-browserstack/pull/1/files#r1565167294

File: android/conf/single.conf.js (hunk 2 above)
Comment (avinash-bharti) [reply android/conf/single.conf.js:7]: https://github.com/browserstack/mocha-appium-app-browserstack/pull/1/files#r1565167294

**Review by rushi04 — APPROVED** [review]

**Comment by innovater21** [comment]
LGTM

**Review by rushi04 — APPROVED** [review]

**Review by rajpatta — APPROVED** [review]

**Pull request merged**